### ISSUE TITLE
Preserve Fish that Cannot be Back-Calculated

### ISSUE THREAD

**Issue by droglenc** [fishR-Core-Team/RFishBC#49]
This message came from a user ...

> I am using RFishBC to age and back calculate some bass otoliths. There were a few fish whose otos were not usable, but I wanted to preserve those length/weight entries throughout my data. I used left_join (instead of inner join) when combining my fish data to the age data, which worked to keep those fish. However, when I used the backCalc function, those fish with NAs for ages went away. Is there an argument to keep those when using the backCalc function? Thanks for any tips you can offer.

This does seem to be the case as the code below demonstrates.

```
library(tidyverse)
library(RFishBC)

# Get some data (same data as in vignette)
data(SMBassWB1,package="RFishBC") ## fish data
data(SMBassWB2,package="RFishBC") ## rad data

# Simplify to 3 fish so we can see what is going on
tmp1 <- filter(SMBassWB1,id %in% c(377,378,379))
tmp2 <- filter(SMBassWB2,id %in% c(377,378,379))

# Create an extra fish with length (tmp1) but no rad
tmp1 <- rbind(tmp1,
              data.frame(id=999,
                         species="SMB",lake="WB",gear="E",
                         yearcap=1990,lencap=225))
# Combine data frames
tmp <- left_join(tmp1,tmp2,by="id")
tmp

## Demonstrate that fish 999 is lost
res1 <- backCalc(tmp,lencap,BCM="DALE",
                 inFormat="wide",digits=0)
res1
```

**Comment by droglenc** [comment]
I don't see a simple way to modify the code to allow this.  These fish are dropped when missing radii are removed (after changing to long format) in the following line ...

```
dat <- dat[!is.na(dat$rad),]
```

This can be addressed by removing all rows that don't have `rad` data but do have `radcap` data (so some measurement was made but not for that annulus). However, this leaves many rows of data for the fish without the radial measurements. These rows can be removed by removing rows with duplicated entries for `id` and `rad`. However, the fish without radial measurements that we are trying to keep are then dropped when removing the plus growth ...

```
if (deletePlusGrowth) dat <- dat[dat$ann<=dat$agecap,]
```

... as the comparison is with `NA` values for this fish. This can probably be addressed as well with some sort of logical but I am worried that we are possible including more issues in the code to try to handle this situation. I offer a work-around in the next comment until we can study this more.

**Comment by droglenc** [comment]
A possible interim solution is to (using the `tmp1` and `tmp2` data frames from the original comment).

1. Get ids that have ages from tmp2

```
( ids_w_ages <- unique(tmp2$id) )
```

2. Separate fish data into those with and without ages

```
( tmp1_wages <- filter(tmp1,id %in% ids_w_ages) )
( tmp1_woages <- filter(tmp1,!id %in% ids_w_ages) )
```

3. Combine rad data with fish data for those with ages

```
( tmp <- inner_join(tmp1_wages,tmp2) )
```

4. back-calculate lengths for those fish

```
( tmp <- backCalc(tmp,lencap,BCM="DALE",
                  inFormat="wide",digits=0) )
```

5. Combine on fish data for those without ages

```
( final <- full_join(tmp1_woages,tmp) )
```

**Comment by droglenc** [comment]
This should be fixed on v0.2.7.
